frontend/src/scenes/billing/billingProductLogic.ts

```diff
@@ -215,5 +228,38 @@ export const billingProductLogic = kea<billingProductLogicType>([
             })
             actions.setSurveyID('')
         },
+        setScrollToProductKey: ({ scrollToProductKey }) => {
+            if (scrollToProductKey && scrollToProductKey === props.product.type) {
+                const { currentPlan } = values.currentAndUpgradePlans
+
+                if (currentPlan.initial_billing_limit) {
+                    actions.setProductSpecificAlert({
+                        status: 'warning',
+                        title: 'Billing Limit Automatically Applied',
+                        message: `To protect your costs and ours, we've automatically applied a $${currentPlan?.initial_billing_limit} billing limit for ${props.product.name}.`,
+                        action: {
+                            onClick: () => {
+                                actions.setIsEditingBillingLimit(true)
+                                setTimeout(() => {
+                                    if (props.billingLimitInputRef?.current) {
+                                        props.billingLimitInputRef?.current.focus()
+                                        props.billingLimitInputRef?.current.scrollIntoView({
+                                            behavior: 'smooth',
+                                            block: 'nearest',
+                                        })
+                                    }
+                                }, 0)
```

Comment:
what's the reason for the timeout here?

Reply:
We need a small delay to ensure that the input is rendered after we call `actions.setIsEditingBillingLimit(true)`.

Reply:
Hm, ok, that seems a bit hacky and prone to errors. We know if the input exists if `props.billingLimitInputRef?.current` is a truthy value. So if not true then we can use a [kea breakpoint](https://keajs.org/docs/core/listeners/#breakpoints) to wait a moment before trying again. That could also be prone to errors.... but it's more understandable than having a timeout with a wait period of 0. 

If you _really_ want to go nuts you could set on the product a value if the input has been rendered and populate it with a useEffect on the ref...... but that seems unnecessary haha

Reply:
Technically, the setTimeout({}, 0) inserts the code to be executed to the end of the taskQueue ([a more in-depth explanation](https://www.justjeb.com/post/why-does-settimeout-work)), allowing whatever else was queued in between the last function call and itself to be run first. I've used it a number of times and have confidence in its functionality. If you still feel it's gnarly, I'm happy to take it out and use the breakpoint.

Reply:
This is Maruis's reasoning for using breakpoints instead of setTimeout: https://posthog.slack.com/archives/C0113360FFV/p1613376298001200?thread_ts=1613374480.000400&cid=C0113360FFV

It does look like we do this with setTimeout in other places, so feel free to do either!